### PR TITLE
Provide autocomplete for Parent, InstanceOf, and obeys

### PR DESCRIPTION
**Pull request by mint-thompson**
Completes task [CIMPL-336](https://standardhealthrecord.atlassian.net/browse/CIMPL-336).

When activating autocomplete after the `Parent` keyword, completion items will include applicable FSH entities. What counts as applicable depends on what kind of entity is currently being defined.

When activating autocomplete after the `InstanceOf` keyword, completion items will include FSH Profiles, Resources, and Extensions.

When activating autocomplete while writing an `ObeysRule`, completion items will include FSH Invariants.

The FSH grammar files are updated based on the SUSHI 2.0.0 release.

The launch configuration present in the repo should be suitable for testing locally, but let me know if you have questions about how to test. You may need to add a second command-line argument to the launch configuration specifying a folder to open, since the "Go To Definition" feature only works when there is an active workspace.

### CODE REVIEW

File: src/FshCompletionProvider.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    } else if (currentLine.startsWith('Parent:')) {
+      // find the most recent non-empty line
+      let previousLineNumber: number;
+      for (previousLineNumber = position.line - 1; previousLineNumber >= 0; previousLineNumber--) {
+        const previousLine = document
+          .getText(new Range(previousLineNumber, 0, previousLineNumber + 1, 0))
+          .trim();
+        if (previousLine.length > 0) {
+          if (previousLine.match(/Profile:/)) {
+            allowedTypes = ['Profile', 'Logical'];
```
Comment (jafeltra) [inline src/FshCompletionProvider.ts:50]: I'm not sure if it's too much of an edge case, but it looks like SUSHI allows a Profile to have an Extension as the parent and a Resource as the parent. Is that something we want to support in the autocomplete as well?

**Review by jafeltra — COMMENTED** [review]
This is a really cool feature and the code makes a lot of sense to me! I just had one small question about the allowed parents on Profiles and two small comments about comments.

**Review by jafeltra — APPROVED** [review]
This is an awesome feature!!

**Review by ngfreiter — APPROVED** [review]
Looks good to me!

**Pull request merged**